scripts/src/javatests/org/oppia/android/scripts/regex/RegexPatternValidationCheckTest.kt

```diff
@@ -2153,6 +2158,98 @@ class RegexPatternValidationCheckTest {
+        app:cardBackgroundColor="@color/color_palette_primary_text_color"
+        app:cardBackgroundColor="#000000"        
+        android:background="@color/component_color_shared_primary_text_color"
+        android:background="@color/color_defs_shared_primary_text_color"
+        android:background="@color/color_palette_primary_text_color"
+        android:background="#000000"
+      """.trimIndent()
+    tempFolder.newFolder("testfiles", "app", "src", "main", "res", "layout")
+    val stringFilePath = "app/src/main/res/layout.*?/.+?\\.xml"
+    tempFolder.newFile("testfiles/$stringFilePath").writeText(prohibitedContent)
+
+    val exception = assertThrows(Exception::class) {
+      runScript()
+    }
+
+    // Verify that all patterns are properly detected & prohibited.
+    assertThat(exception).hasMessageThat().contains(REGEX_CHECK_FAILED_OUTPUT_INDICATOR)
+    assertThat(outContent.toString().trim())
+      .isEqualTo(
+        """
+        $stringFilePath:2: $doesNotReferenceColorFromComponentColorInLayouts
+        $stringFilePath:3: $doesNotReferenceColorFromComponentColorInLayouts
+        $stringFilePath:4: $doesNotReferenceColorFromComponentColorInLayouts
+        $stringFilePath:6: $doesNotReferenceColorFromComponentColorInLayouts
+        $stringFilePath:7: $doesNotReferenceColorFromComponentColorInLayouts
+        $stringFilePath:8: $doesNotReferenceColorFromComponentColorInLayouts
+        $stringFilePath:10: $doesNotReferenceColorFromComponentColorInLayouts
+        $stringFilePath:11: $doesNotReferenceColorFromComponentColorInLayouts
+        $stringFilePath:12: $doesNotReferenceColorFromComponentColorInLayouts
+        $stringFilePath:14: $doesNotReferenceColorFromComponentColorInLayouts
+        $stringFilePath:15: $doesNotReferenceColorFromComponentColorInLayouts
+        $stringFilePath:16: $doesNotReferenceColorFromComponentColorInLayouts
+        $stringFilePath:18: $doesNotReferenceColorFromComponentColorInLayouts
+        $stringFilePath:19: $doesNotReferenceColorFromComponentColorInLayouts
+        $stringFilePath:20: $doesNotReferenceColorFromComponentColorInLayouts
+        $stringFilePath:22: $doesNotReferenceColorFromComponentColorInLayouts
+        $stringFilePath:23: $doesNotReferenceColorFromComponentColorInLayouts
+        $stringFilePath:24: $doesNotReferenceColorFromComponentColorInLayouts
+        $wikiReferenceNote
+        """.trimIndent()
+      )
+  }
+
+  fun testFileContent_kotlinFiles_includesNonColorComponentReferences_fileContentIsNotCorrect() {
+    val prohibitedContent =
+      """
+        R.color.component_color_shared_activity_status_bar_color
+        R.color.color_def_avatar_background_1
+        R.color.color_palette_primary_color
+      """.trimIndent()
+    tempFolder.newFolder("testfiles", "app", "src", "main", "java", "org", "oppia", "android")
+    val stringFilePath = "app/src/main/java/org/oppia/android/app.?/." +
```

Comment:
Does this actually work? I didn't think newFile() works with patterns.

I think for this case we should be creating an example file for each type that should be prohibited (which means that multiple filepaths will be present in the failure text).

Reply:
How I can create example file? Is that manually like we add new file or is there any script through test itself create one?

Reply:
Latest changes have addressed this.